cdap-ui/app/directives/flow-graph/flow.js

```diff
@@ -54,6 +64,7 @@ module.directive('myFlowGraph', function (d3, dagreD3, $state, $filter) {
         angular.forEach(nodes, function (node) {
           var nodeLabel = node.name.length > 8 ? node.name.substr(0, 5) + '...' : node.name;
           instanceMap[node.name] = node;
+          labelMap[nodeLabel] = node;
           if (node.type === 'STREAM') {
```

Comment:
Same as the other comment. In case of workflows we have the node type as 'ACTION' or 'FORK'..